### DISCUSSION TITLE
Unit Tests that aren't run, run

### DISCUSSION THREAD

**Discussion by tig** [tui-cs/Terminal.Gui#2602] [General]
@BDisp has commented about this too.

It is freaking driving me nuts!

Even if I just run a subset of the unit tests in Test Explorer, in the background it seems others are running, including the ones that set/get the clipboard!! I'm constantly finding my clipboard is full of test text.

Anyone know why this is happening?

**Comment by BDisp** [comment]
Ah I though that only happening to me. @tig this situation only happening after we finished running all the unit test, which is normal because the Clipboard unit test was runs. After you refactored the Clipboard a long time ago this is happening even after running a single unit test without Clipboard involved. It seems that the auto-init is starting some background thread related with the Clipboard. Many times I copy anything from the code and when I'm going to paste, isn't what I latest copied. This is very frustrating because it repeats over and over forcing me to copy again to being able to paste.

**Comment by tig** [comment]
This may be it. Trying with FCC disabled....

https://stackoverflow.com/questions/70231309/xunit-nunit-msunit-all-run-all-tests-twice-vs2022

**Comment by BDisp** [comment]
I also will try `"What seems to work is to enable RunMsCodeCoverage in the Fine Code Coverage settings."` first.
